// to-plus-frontend/src/components/BrandLogo.tsx

import React from 'react';

interface BrandLogoProps {
  size?: 'sm' | 'md' | 'lg' | 'xl';
  variant?: 'full' | 'icon' | 'text';
  className?: string;
  light?: boolean; // Force white text (for dark backgrounds)
  forceDarkText?: boolean; // Force dark text (for light backgrounds even in dark mode)
}

const BrandLogo: React.FC<BrandLogoProps> = ({ size = 'md', variant = 'full', className = '', light = false, forceDarkText = false }) => {
  
  // Size Configuration
  const sizes = {
    sm: { h: 24, w: 24, text: 'text-lg', gap: 'gap-2', subtext: 'text-[8px]' },
    // MD size optimized for Sidebar
    md: { h: 32, w: 32, text: 'text-2xl', gap: 'gap-3', subtext: 'text-[10px]' }, 
    lg: { h: 48, w: 48, text: 'text-4xl', gap: 'gap-4', subtext: 'text-xs' },
    xl: { h: 64, w: 64, text: 'text-5xl', gap: 'gap-5', subtext: 'text-sm' },
  };

  const currentSize = sizes[size];
  
  // Color Logic
  let textColorClass = 'text-slate-900 dark:text-white';
  let subTextColorClass = 'text-slate-500 dark:text-slate-400';

  if (light) {
    textColorClass = 'text-white';
    subTextColorClass = 'text-slate-400';
  } else if (forceDarkText) {
    textColorClass = 'text-slate-900';
    subTextColorClass = 'text-slate-500';
  }

  const brandColorClass = light ? 'text-brand-400' : 'text-brand-600';
  
  // Custom SVG: Blue Rounded Cross
  const LogoIcon = () => (
    <svg 
      width={currentSize.w} 
      height={currentSize.h} 
      viewBox="0 0 100 100" 
      fill="none" 
      xmlns="http://www.w3.org/2000/svg"
      className="shrink-0"
    >
      {/* Vertical Bar */}
      <rect x="38" y="0" width="24" height="100" rx="12" className={light ? 'fill-brand-400' : 'fill-brand-600'} />
      {/* Horizontal Bar */}
      <rect x="0" y="38" width="100" height="24" rx="12" className={light ? 'fill-brand-400' : 'fill-brand-600'} />
    </svg>
  );

  return (
    <div className={`flex items-center ${currentSize.gap} font-sans select-none ${className}`}>
      {(variant === 'full' || variant === 'icon') && <LogoIcon />}
      
      {(variant === 'full' || variant === 'text') && (
        <div className="flex flex-col justify-center">
          <div className={`leading-none ${currentSize.text} flex items-baseline`}>
            {/* "To" uses determined color class */}
            <span className={`font-bold tracking-tight ${textColorClass}`}>To</span>
            <span className={`font-black ${brandColorClass} ml-0.5`}>ERP</span>
            <span className={`font-medium ${textColorClass} ml-1.5 opacity-90`}>Bulut</span>
          </div>
          {size !== 'sm' && (
            <span className={`${currentSize.subtext} font-semibold tracking-wide ${subTextColorClass} mt-0.5 leading-none`}>
              Dijital İş Zekası
            </span>
          )}
        </div>
      )}
    </div>
  );
};

export default BrandLogo;
