// to-plus-backend/dist/prisma/seed.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const client_1 = require("@prisma/client");
const bcrypt = require("bcrypt");
const prisma = new client_1.PrismaClient();
const data = {
    "packages": [
        {
            "id": "pkg_starter",
            "name": "Başlangıç Paketi",
            "priceMonthly": 199,
            "priceYearly": 1990,
            "maxUsers": 3,
            "storageLimit": "5GB",
            "isPopular": false,
            "features": ["Temel Modüller", "3 Kullanıcı", "5GB Alan"],
            "createdAt": new Date()
        },
        {
            "id": "pkg_pro",
            "name": "Profesyonel Paket",
            "priceMonthly": 399,
            "priceYearly": 3990,
            "maxUsers": 10,
            "storageLimit": "50GB",
            "isPopular": true,
            "features": ["Tüm Modüller", "10 Kullanıcı", "50GB Alan", "E-Fatura"],
            "createdAt": new Date()
        }
    ],
    "tenants": [
        {
            "id": "tenant_system",
            "tag": "system",
            "name": "To ERP Bulut",
            "type": "corporate",
            "subscriptionPlanId": null,
            "status": "active"
        },
        {
            "id": "tenant_demo_1",
            "tag": "demo-firma",
            "name": "ToDestek Bilişim",
            "type": "corporate",
            "subscriptionPlanId": "pkg_pro",
            "status": "active"
        }
    ],
    "users": [
        {
            "id": "usr_super_admin",
            "tenantId": "tenant_system",
            "name": "Sistem Sahibi",
            "email": "admin@tobulut.com",
            "passwordHash": "hashed_password123",
            "role": "superuser",
            "isSuperAdmin": true,
            "permissions": {}
        },
        {
            "id": "usr_admin_1",
            "tenantId": "tenant_demo_1",
            "name": "Ünal Uslusoy (Yönetici)",
            "email": "unaluslusoy@todestek.net",
            "passwordHash": "hashed_password123",
            "role": "admin",
            "isSuperAdmin": false,
            "permissions": {}
        },
        {
            "id": "usr_accountant_1",
            "tenantId": "tenant_demo_1",
            "name": "Zeynep Kaya (Muhasebe)",
            "email": "zeynep@todestek.com",
            "passwordHash": "hashed_password123",
            "role": "accountant",
            "isSuperAdmin": false,
            "permissions": {}
        },
        {
            "id": "usr_tech_1",
            "tenantId": "tenant_demo_1",
            "name": "Ali Veli (Tekniker)",
            "email": "ali@todestek.com",
            "passwordHash": "hashed_password123",
            "role": "technician",
            "isSuperAdmin": false,
            "permissions": {}
        }
    ],
    "accounts": [
        {
            "id": "acc_cust_1",
            "tenantId": "tenant_demo_1",
            "accountCode": "120.01.001",
            "type": "customer",
            "name": "Mega İnşaat Ltd. Şti.",
            "balance": 15400.00
        },
        {
            "id": "acc_supp_1",
            "tenantId": "tenant_demo_1",
            "accountCode": "320.01.005",
            "type": "supplier",
            "name": "TeknoTedarik A.Ş.",
            "balance": -45000.00
        }
    ],
    "products": [
        {
            "id": "prd_1",
            "tenantId": "tenant_demo_1",
            "code": "LT-2024",
            "name": "Laptop Pro X1 - i7 16GB",
            "stockQuantity": 12,
            "priceSell": 24500.00,
            "trackStock": true
        },
        {
            "id": "prd_2",
            "tenantId": "tenant_demo_1",
            "code": "MS-500",
            "name": "Kablosuz Mouse",
            "stockQuantity": 85,
            "priceSell": 450.00,
            "trackStock": true
        },
        {
            "id": "prd_srv_1",
            "tenantId": "tenant_demo_1",
            "code": "SRV-FORMAT",
            "name": "Format ve Kurulum Hizmeti",
            "stockQuantity": 9999,
            "priceSell": 750.00,
            "trackStock": false
        }
    ],
    "cashRegisters": [
        {
            "id": "reg_main_tl",
            "tenantId": "tenant_demo_1",
            "name": "Merkez Kasa (TL)",
            "currency": "TRY",
            "balance": 12500.50
        },
        {
            "id": "reg_bank_garanti",
            "tenantId": "tenant_demo_1",
            "name": "Garanti Bankası",
            "currency": "TRY",
            "balance": 154000.00
        }
    ],
    "invoices": [
        {
            "id": "inv_1",
            "tenantId": "tenant_demo_1",
            "accountId": "acc_cust_1",
            "type": "sales",
            "status": "paid",
            "totalAmount": 58800.00,
            "items": [
                {
                    "productId": "prd_1",
                    "name": "Laptop Pro X1 - i7 16GB",
                    "quantity": 2,
                    "price": 24500.00,
                    "total": 49000.00
                }
            ]
        }
    ],
    "transactions": [
        {
            "id": "trx_1",
            "tenantId": "tenant_demo_1",
            "registerId": "reg_bank_garanti",
            "accountId": "acc_cust_1",
            "type": "income",
            "amount": 58800.00,
            "description": "Fatura Tahsilatı - GIB2024000001",
            "invoiceId": "inv_1"
        },
        {
            "id": "trx_2",
            "tenantId": "tenant_demo_1",
            "registerId": "reg_main_tl",
            "type": "expense",
            "amount": 450.00,
            "description": "Ofis Kırtasiye Gideri"
        }
    ],
    "serviceTickets": [
        {
            "id": "srv_1",
            "tenantId": "tenant_demo_1",
            "customerId": "acc_cust_1",
            "technicianId": null,
            "deviceInfo": { "device": "HP Plotter T200", "issue": "Kağıt sıkıştırıyor, baskı kafası temizliği gerekli." },
            "status": "processing",
            "estimatedCost": 1500.00
        }
    ]
};
async function main() {
    console.log('Start seeding ...');
    for (const pkg of data.packages) {
        await prisma.subscriptionPackage.upsert({
            where: { id: pkg.id },
            update: {
                priceYearly: pkg.priceYearly,
                maxUsers: pkg.maxUsers,
                storageLimit: pkg.storageLimit,
                features: pkg.features,
                isPopular: pkg.isPopular
            },
            create: {
                id: pkg.id,
                name: pkg.name,
                priceMonthly: pkg.priceMonthly,
                priceYearly: pkg.priceYearly,
                maxUsers: pkg.maxUsers,
                storageLimit: pkg.storageLimit,
                features: pkg.features,
                isPopular: pkg.isPopular,
                isActive: true
            },
        });
    }
    for (const t of data.tenants) {
        await prisma.tenant.upsert({
            where: { id: t.id },
            update: {
                name: t.name,
                tag: t.tag,
                type: t.type
            },
            create: {
                id: t.id,
                tag: t.tag,
                name: t.name,
                type: t.type,
                subscriptionPlanId: t.subscriptionPlanId,
                status: 'active'
            },
        });
    }
    const passwordHash = await bcrypt.hash('password123', 10);
    for (const u of data.users) {
        await prisma.user.upsert({
            where: { id: u.id },
            update: {
                passwordHash: passwordHash,
                name: u.name,
                email: u.email,
                role: u.role,
                isSuperAdmin: u.isSuperAdmin || false
            },
            create: {
                id: u.id,
                tenantId: u.tenantId,
                name: u.name,
                email: u.email,
                passwordHash: passwordHash,
                role: u.role,
                permissions: u.permissions,
                isSuperAdmin: u.isSuperAdmin || false
            }
        });
    }
    for (const acc of data.accounts) {
        await prisma.account.upsert({
            where: { id: acc.id },
            update: {},
            create: {
                id: acc.id,
                tenantId: acc.tenantId,
                name: acc.name,
                accountCode: acc.accountCode,
                type: acc.type,
                balance: acc.balance
            }
        });
    }
    for (const p of data.products) {
        await prisma.product.upsert({
            where: { id: p.id },
            update: {},
            create: {
                id: p.id,
                tenantId: p.tenantId,
                name: p.name,
                code: p.code,
                stockQuantity: p.stockQuantity,
                priceSell: p.priceSell,
                trackStock: p.trackStock
            }
        });
    }
    for (const cr of data.cashRegisters) {
        await prisma.cashRegister.upsert({
            where: { id: cr.id },
            update: {},
            create: {
                id: cr.id,
                tenantId: cr.tenantId,
                name: cr.name,
                currency: cr.currency,
                balance: cr.balance
            }
        });
    }
    for (const inv of data.invoices) {
        const exists = await prisma.invoice.findUnique({ where: { id: inv.id } });
        if (!exists) {
            await prisma.invoice.create({
                data: {
                    id: inv.id,
                    tenantId: inv.tenantId,
                    accountId: inv.accountId,
                    type: inv.type,
                    status: inv.status,
                    totalAmount: inv.totalAmount,
                    items: {
                        create: inv.items.map(item => ({
                            productId: item.productId,
                            name: item.name || "",
                            quantity: item.quantity,
                            price: item.price,
                            total: item.total
                        }))
                    }
                }
            });
        }
    }
    for (const trx of data.transactions) {
        const exists = await prisma.transaction.findUnique({ where: { id: trx.id } });
        if (!exists) {
            await prisma.transaction.create({
                data: {
                    id: trx.id,
                    tenantId: trx.tenantId,
                    registerId: trx.registerId,
                    accountId: trx.accountId,
                    type: trx.type,
                    amount: trx.amount,
                    description: trx.description,
                    invoiceId: trx.invoiceId
                }
            });
        }
    }
    for (const srv of data.serviceTickets) {
        const exists = await prisma.serviceTicket.findUnique({ where: { id: srv.id } });
        if (!exists) {
            await prisma.serviceTicket.create({
                data: {
                    id: srv.id,
                    tenantId: srv.tenantId,
                    customerId: srv.customerId,
                    deviceInfo: srv.deviceInfo,
                    status: srv.status,
                    estimatedCost: srv.estimatedCost
                }
            });
        }
    }
    console.log('Seeding finished.');
}
main()
    .catch((e) => {
    console.error(e);
    process.exit(1);
})
    .finally(async () => {
    await prisma.$disconnect();
});
