// to-plus-backend/dist/src/app.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppModule = void 0;
const common_1 = require("@nestjs/common");
const config_1 = require("@nestjs/config");
const throttler_1 = require("@nestjs/throttler");
const prisma_service_1 = require("./prisma/prisma.service");
const auth_module_1 = require("./modules/auth/auth.module");
const products_module_1 = require("./modules/products/products.module");
const accounts_module_1 = require("./modules/accounts/accounts.module");
const finance_module_1 = require("./modules/finance/finance.module");
const invoices_module_1 = require("./modules/invoices/invoices.module");
const services_module_1 = require("./modules/services/services.module");
const hr_module_1 = require("./modules/hr/hr.module");
const saas_module_1 = require("./modules/saas/saas.module");
const webhooks_module_1 = require("./modules/webhooks/webhooks.module");
const reports_module_1 = require("./modules/reports/reports.module");
const users_module_1 = require("./modules/users/users.module");
const tasks_module_1 = require("./modules/tasks/tasks.module");
const inventory_module_1 = require("./modules/inventory/inventory.module");
const offers_module_1 = require("./modules/offers/offers.module");
const sales_module_1 = require("./modules/sales/sales.module");
const campaigns_module_1 = require("./modules/campaigns/campaigns.module");
const notifications_module_1 = require("./modules/notifications/notifications.module");
const admin_module_1 = require("./modules/admin/admin.module");
const uploads_module_1 = require("./modules/uploads/uploads.module");
const super_admin_module_1 = require("./modules/super-admin/super-admin.module");
let AppModule = class AppModule {
};
exports.AppModule = AppModule;
exports.AppModule = AppModule = __decorate([
    (0, common_1.Module)({
        imports: [
            config_1.ConfigModule.forRoot({ isGlobal: true }),
            throttler_1.ThrottlerModule.forRoot([{
                    ttl: 60000,
                    limit: 10,
                }]),
            auth_module_1.AuthModule,
            products_module_1.ProductsModule,
            accounts_module_1.AccountsModule,
            finance_module_1.FinanceModule,
            invoices_module_1.InvoicesModule,
            services_module_1.ServicesModule,
            hr_module_1.HrModule,
            saas_module_1.SaasModule,
            webhooks_module_1.WebhooksModule,
            reports_module_1.ReportsModule,
            users_module_1.UsersModule,
            tasks_module_1.TasksModule,
            inventory_module_1.InventoryModule,
            offers_module_1.OffersModule,
            sales_module_1.SalesModule,
            campaigns_module_1.CampaignsModule,
            notifications_module_1.NotificationsModule,
            notifications_module_1.NotificationsModule,
            admin_module_1.AdminModule,
            uploads_module_1.UploadsModule,
            super_admin_module_1.SuperAdminModule,
        ],
        controllers: [],
        providers: [prisma_service_1.PrismaService],
        exports: [prisma_service_1.PrismaService],
    })
], AppModule);
